test/fw/ptl/utils/plugins/ptl_report_json.py

```diff
@@ -0,0 +1,155 @@
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+import time
+import os
+import sys
+from ptl.utils.pbs_dshutils import DshUtils
+
+
+class PTLJsonData(object):
+    """
+    The intent of the class is to generate json format of PTL test data
+    """
+
+    def __init__(self):
+        self.__du = DshUtils()
+        self.__cmd = [os.path.basename(sys.argv[0])]
+        self.__cmd += sys.argv[1:]
+
+    def get_json(self, data=None, prev_data=None):
+        """
+        Method to generate test data in accordance to json schema
+
+        param data: dictionary of a test case details
+        type data: dict
+
+        returns a formatted dictionary of the data
+        """
+        data_json = None
+        if not prev_data:
+            data_json = {
+                'command': ' '.join(self.__cmd),
+                'user': self.__du.get_current_user(),
+                'product_version': data['pbs_version'],
+                'run_id': int(time.time()),
+                'test_conf': {},
+                'machine_info': data['machinfo'],
+                'testsuites': {},
+                'additional_data': {},
+                'test_summary': {
+                    'result_summary': {
+                        'run': 0,
+                        'succeeded': 0,
+                        'failed': 0,
+                        'errors': 0,
+                        'skipped': 0,
+                        'timedout': 0
+                    },
+                    'test_start_time': str(data['start_time']),
+                    'tests_with_failures': [],
+                    'test_suites_with_failures': []
+                }
+            }
+            if data['testparam']:
+                for param in data['testparam'].split(','):
+                    par = param.split('=', 1)
+                    data_json['test_conf'][par[0]] = par[1]
+        else:
+            data_json = prev_data
```

Comment:
Also, just making sure, this will make both 'data_json' and 'prev_data' point to the same memory, so anything you modify for 'data_json' will also change 'prev_data', that won't cause any issues right?

Reply:
data_json['test_start_time'] is the start time of the whole run. Hence only needs to be assigned value at the first test case run i.e. in line 83.
Yes, there wont be any issues since prev_data is every time extension of data_json which is dumped to json file; as and when it gets updated.